### PR TITLE
Persistent caching for interactive credentials on Linux and macOS

### PR DESCRIPTION
**Pull request by chlowell**
This adopts msal-extensions 0.2.2 to enable persistent token caching on Linux and macOS for `DeviceCodeCredential` and `InteractiveBrowserCredential`. It also adds a new keyword argument for those credentials, `allow_unencrypted_cache`, which determines whether to fall back to a plaintext cache on Linux when libsecret or PyGObject are unavailable (defaults `False`).

Strings and paths for the new platforms are from Azure/azure-sdk-for-java#9188 and the default behavior of [MSAL's Java extension library](https://github.com/AzureAD/microsoft-authentication-extensions-for-java/tree/dev/src/main/java/com/microsoft/aad/msal4jextensions).

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py [hunk 1]
```diff
@@ -105,7 +91,8 @@ def __init__(self, client_id, client_credential=None, **kwargs):
         self._cache = kwargs.pop("_cache", None)  # internal, for use in tests
         if not self._cache:
             if kwargs.pop("enable_persistent_cache", False):
-                self._cache = _load_persistent_cache()
+                allow_unencrypted = kwargs.pop("allow_unencrypted_cache", False)
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py:94]: Is this a public kwarg?

I don't see it in docstring.

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py:94]: Yes. Check the diffs for browser.py and device_code.py.

Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py:94]: Do you mean users are not supposed to use MsalCredential directly?

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py:94]: Yes.

File: sdk/identity/azure-identity/azure/identity/_internal/persistent_cache.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            )
+        except ImportError:
+            if not allow_unencrypted:
+                raise ValueError(
+                    "PyGObject is required to encrypt the persistent cache. Please install that library or ",
+                    "specify 'allow_unencrypted_cache=True' to store the cache without encryption.",
+                )
+            persistence = msal_extensions.FilePersistence(file_path)
+    else:
+        raise NotImplementedError("A persistent cache is not available on this platform.")
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_internal/persistent_cache.py:51]: Do you think we should give more accurate error message if it is used on windows by "LOCALAPPDATA" is not set?

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_internal/persistent_cache.py:51]: What do you have in mind for a more accurate message? Something like "Environment variable LOCALAPPDATA isn't set"?

Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_internal/persistent_cache.py:51]: Yes. If I see "A persistent cache is not available on this platform" on windows, I will feel very confused. :)

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_internal/persistent_cache.py]: I agree with you there. I'm content with simply saying the cache is unavailable in this case but I think saying it's unavailable *on this platform* is misleading. I changed the message to say the cache is unavailable *in this environment* instead.

**Pull request merged**